maestro-android/src/main/java/dev/mobile/maestro/receivers/LocaleSettingReceiver.kt

```diff
@@ -0,0 +1,105 @@
+package dev.mobile.maestro.receivers
+
+import android.app.Activity
+import android.content.BroadcastReceiver
+import android.content.Context
+import android.content.Intent
+import android.util.Log
+import dev.mobile.maestro.handlers.LocaleSettingHandler
+import org.apache.commons.lang3.LocaleUtils
+import java.util.*
+
+class LocaleSettingReceiver : BroadcastReceiver(), HasAction {
+    override fun onReceive(context: Context?, intent: Intent?) {
+        var language = intent?.getStringExtra(LANG)
+        var country = intent?.getStringExtra(COUNTRY)
+
+        if (language == null || country == null) {
+            Log.w(TAG, "It is required to provide both language and country, for example: " +
+                    "am broadcast -a dev.mobile.maestro --es lang ja --es country JP")
+            Log.i(TAG, "Set en-US by default.")
+            language = "en"
+            country = "US"
+        }
+
+        var locale = Locale(language, country)
+
+        Log.i(TAG, "Obtained locale: $locale")
+
+        val script = intent?.getStringExtra(SCRIPT)
+        if (script != null) {
+            Locale.Builder().setLocale(locale).setScript(script).build().also { locale = it }
+        }
+
+        if (!LocaleUtils.isAvailableLocale(locale)) {
+            val approximateMatchesLc = matchLocales(language, country)
+
+            if (approximateMatchesLc.isNotEmpty() && script.isNullOrBlank()) {
+                Log.i(
+                    TAG,
+                    "The locale $locale is not known. Selecting the closest known one ${approximateMatchesLc[0]} instead"
```

Comment:
How will we surface this to users so they know that something other than what they specified was used?

Reply:
looking at Appium they do some soft handling here, by picking up the closest locale possible if Locale() object is failed to be initialized, I think I'm also fine with this approach, in the end we don't have a control of typos / not supported locales for specific Android versions.